quicklogic/common/toolchain_wrappers/ql_symbiflow

```diff
@@ -0,0 +1,459 @@
+			DEVICENAME=0
+			PARTNAME=0
+			SDCNAME=0
+			OUTNAME=0
+			ROUTE_NAME0=1
+			;;
+		-dump)
+			VERILOGLIST=0
+			PCFNAME=0
+			TOPNAME=0
+			SOURCEDIR=0
+			HELPNAME=0
+			DEVICENAME=0
+			PARTNAME=0
+			SDCNAME=0
+			OUTNAME=1
+			ROUTE_NAME0=0
+			;;
+		-synth|-compile)
+			VERILOGLIST=0
+			PCFNAME=0
+			TOPNAME=0
+			SOURCEDIR=0
+			HELPNAME=0
+			DEVICENAME=0
+			PARTNAME=0
+			SDCNAME=0
+			OUTNAME=0
+			ROUTE_NAME0=0
+			;;
+
+		*)
+			if [ $SOURCEDIR -eq 1 ]; then
+				SOURCE=$arg_v
+			elif [ $VERILOGLIST -eq 1 ]; then
+				VERILOG_FILES+="$arg_v "
+			elif [ $HELPNAME -eq 1 ]; then
+				echo "" 
+			elif [ $TOPNAME -eq 1 ]; then
+				TOP=$arg_v
+			elif [ $DEVICENAME -eq 1 ]; then
+				DEVICE="${arg_v}"
+				#DEVICE="${DEVICE,,}"
+            	case ${DEVICE} in
+            		qlf_k4n8)
+				DEVICE="${DEVICE}_${DEVICE}"
+            			FAMILY="qlf_k4n8"
+            			;;
+            		*)
+            			echo "Unsupported device '${DEVICE}'"
+            			exit 1
+            			;;
+            	esac
+			elif [ $PCFNAME -eq 1 ]; then
+				PCF="$arg_v"
+			elif [ $SDCNAME -eq 1 ]; then
+				SDC="$arg_v"
+			elif [ $ROUTE_NAME0 -eq 1 ]; then
+				ROUTE_FLAG0="$arg_v"
+				ROUTE_FLAG0="${ROUTE_FLAG0,,}"
+			elif [ $OUTNAME -eq 1 ]; then
+				OUT+="$arg_v "
+			elif [ $PARTNAME -eq 1 ]; then
+				PART="${arg_v}"
+ 				PART="${PART^^}"
+			else
+				echo "Refer help for more details: ql_symbiflow -h "
+                                exit 1
+
+			fi
+			;;
+	esac
+done
+
+
+##### Check if the source directory exists #####
+if [[ $1 == "-h" || $1 == "--help" ]];then
+    exit 1
+else
+    if [ -z "$SOURCE" ];then
+    SOURCE=$PWD
+    elif [ $SOURCE == "." ];then
+    SOURCE=$PWD
+    elif [ ! -d "$SOURCE" ];then
+    echo "Directory path $SOURCE DOES NOT exists. Please add absolute path" 
+    exit 1
+    fi
+
+if [[ $1 == "-h"  ||  $1 == "--help" ]];then
+exit 0
+else
+if [ -f $SOURCE/v_list_tmp ];then 
+rm -f $SOURCE/v_list_tmp 
+fi
+echo "$VERILOG_FILES" >${SOURCE}/v_list
+fi
+
+##### Validate the verlog source files #####
+
+if [ ${#VERILOG_FILES[@]} -eq 0 ]; then
+       if [[ $1 != "-h" || $1 != "--help" ]];then
+	echo "Please provide at least one Verilog file"
+	exit 1
+       fi
+else
+	echo "verilog files: $VERILOG_FILES"
+	echo $VERILOG_FILES >${SOURCE}/v_list
+	sed '/^$/d' $SOURCE/v_list > $SOURCE/f_list_temp
+	VERILOG_FILES=`cat $SOURCE/f_list_temp`
+fi
+fi
+
+if [[ $1 == "-compile" || $1 == "-post_verilog" ]]; then
+  # FIXME: Allow no PCF/pinmap for some devices
+  if [[ ! "$DEVICE" =~ ^(qlf_k4n8_qlf_k4n8)$ ]]; then
+    if [ -z "$PCF" ]; then
+      echo "PCF file option is missing. Refer -h/--help"
+      exit 1
+    elif ! [ -f "$SOURCE/$PCF" ]; then
+      echo "The pcf file: $PCF is missing at: $SOURCE"
+      exit 1
+    fi
+  fi
+  if [ -z "$DEVICE" ]; then
+    echo "DEVICE name is missing. Refer -h/--help"
+    exit 1
+  elif ! [[ "$DEVICE" =~ ^(ql-eos-s3_wlcsp|ql-ap3-ql745a_ql745a|qlf_k4n8_qlf_k4n8)$ ]]; then 
```

Comment:
why do we check for eos-s3 and ap3? those are not supported by this toolchain

Reply:
Right, removed all the reference to eos-s3 and ap3.